ui/dashboards/src/components/Panel/Panel.tsx

```diff
@@ -17,28 +17,41 @@ import { useInView } from 'react-intersection-observer';
 import { ErrorBoundary, ErrorAlert, combineSx, useId, useChartsTheme } from '@perses-dev/components';
 import { PanelDefinition } from '@perses-dev/core';
 import { Card, CardProps, CardContent } from '@mui/material';
+import { PanelGroupItemId } from '../../context';
 import { PanelHeader, PanelHeaderProps } from './PanelHeader';
 import { PanelContent } from './PanelContent';
 
 export interface PanelProps extends CardProps<'section'> {
   definition: PanelDefinition;
   editHandlers?: PanelHeaderProps['editHandlers'];
   panelOptions?: PanelOptions;
+  panelGroupItemId?: PanelGroupItemId;
 }
 
 export type PanelOptions = {
   /**
    * Content to render in the top-right corner of the panel. It will only be
    * rendered when the panel is in edit mode.
    */
-  extra?: () => React.ReactNode;
+  extra?: (props: PanelExtraProps) => React.ReactNode;
+};
+
+export type PanelExtraProps = {
+  /**
+   * The PanelDefinition for the panel.
+   */
+  panelDefinition?: PanelDefinition;
+  /**
+   * The PanelGroupItemId for the panel.
+   */
+  panelGroupItemId?: PanelGroupItemId;
```

Comment:
Yes so it is dashboard specific but its optional. So if its present, it means you're under the dashboard context. By providing the ID, consumers can do a lot more with the panel in the context of the dashboard. Ideally, we add a unique opaque string panel id instead of an object. However, that requires a larger refactor. 

The `Panel` as it stands is a implementation detail of the `Dashboard`. So it really shouldn't be used independently until its properly refactored out the `dashboards` package.